src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -124,7 +142,22 @@ class AuthScreens extends React.Component {
         this.interval = null;
     }
 
+    /**
+     * Setting the initial params would update the URL in the address bar to correctly
+     * It would also setup correct initial state for the Home screen
+     *
+     * @param {String} reportID
+     */
+    setInitialHomeParams(reportID) {
+        this.initialHomeParams = {
+            screen: 'Report',
+            params: {reportID},
+        };
+    }
+
     render() {
+        if (!this.initialHomeParams) { return null; }
```

Comment:
We might replace this with a proper loader here if the overall approach seem good. Doesn't seem necessary as even waiting for the fetch happens very fast